### ISSUE TITLE
Django Styleguide Survey: Feedback

### ISSUE THREAD

**Issue by RadoRado** [HackSoftware/Django-Styleguide#122]
### Discussed in https://github.com/HackSoftware/Django-Styleguide/discussions/114

<div type='discussions-op-text'>

<sup>Originally posted by **RadoRado** January 12, 2022</sup>
A collection if things to address, based on the feedback from the survey we are running - https://www.hacksoft.io/blog/django-styleguide-survey:

- [x] Explain why we are approaching this with a service layer & not custom model methods / custom managers & querysets
- [x] Add reasoning about the lack of ModelViewSet / other DRF generics
- [x] When to use function-based or class-based services
- [x] Make it clear, that you can cherry-pick things that you like from this particular styleguide. By no means you should follow it entirely.
- [ ]  Explain why we are not following strict REST
- [ ] Working with M2M fields (services, selectors) :thinking: 
- [ ] pre-commit hooks :thinking: 
- [ ] Non-API perspective (classic views + templates)
- [ ] Async :thinking: 
- [ ] Recommend default test runner (pytest)
- [ ] Check https://github.com/kiwicom/pytest-recording / https://github.com/betamaxpy/betamax
- [ ] Add alternative to the "giant test folder" from here - https://github.com/HackSoftware/Django-Styleguide#naming-conventions (test per module, as in JavaScript world)
- [ ] Explain API error formatting better
- [ ] Explain the combination of 12factor app & settings files.
- [ ] Add section about i18n/l10
- [ ] Beginner tutorials - how to get from the official Django tutorial to Django Styleguide
- [ ] Add a dedicated section about signals
- [ ] Add a dedicated section about management commands (for example - doing some kind of data migration outside a data migration)
- [ ] Maybe some tips and tricks about deployment part of Django projects and further support. 
- [ ] API versioning
- [ ] Permission Checks / permissions 
- [ ] Mention database transactions
- [ ] Mention DX
- [ ] Mention caching / redis
- [ ] Storing and handling constants
- [ ] Dealing with 3rd party services
- [ ] API Schema / Documentation
- [ ] Use better documentation format (PDF, readthedocs, etc.)
- [ ] Mention graphql</div>

**Comment by RadoRado** [comment]
Original issue that was converted to a discussion & now back to an issue - https://github.com/HackSoftware/Django-Styleguide/issues/90

**Comment by RadoRado** [comment]
Testing issue comment

**Comment by HomaDev** [comment]
Can I contribute? 
I mean would you consider checking PR, or is this repo for HackSoftware employees only?

**Comment by RadoRado** [comment]
@HomaDev Yes, we are open for contributions. I just want to manage expectations - if we decide that the contribution does not sit / fit well with the Styleguide, we might just refer it as an option.

Cheers!

**Comment by HomaDev** [comment]
@RadoRado Thanks! 
I'll try to make something worthy
